src/file_proxy_s3.cc

```diff
@@ -297,17 +323,17 @@ s3_delete_directory(const char* dir_path)
 
 	DeleteObjectsBuffer del_buffer(client, path.GetBucket());
 
-	Aws::S3::Model::ListObjectsRequest req;
+	Aws::S3::Model::ListObjectsV2Request req;
 	req.SetBucket(path.GetBucket());
 	req.SetPrefix(path.GetKey());
 
-	Aws::S3::Model::ListObjectsOutcome res = client.ListObjects(req);
-	if (!res.IsSuccess()) {
-		err("%s", res.GetError().GetMessage().c_str());
+	Aws::Vector<Aws::S3::Model::Object> res;
+	bool success = ListAllObjects(client, req, res);
+	if (!success) {
```

Comment:
Just a suggestion. Do we need the variable for later?  It seems more inline with Clayton's style to use the func as a condition. Your call 
```suggestion
	if (!ListAllObjects(client, req, res)) {
```

Reply:
Agreed, changed all instances.